src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,56 @@
+import { useSearchParams } from 'react-router-dom';
+import { getSearchWith } from '../utils/searchHelper';
+import { SearchLink } from './SearchLink';
+import classNames from 'classnames';
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const sex = searchParams.get('sex') || '';
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+
+  const centuriesArr = ['16', '17', '18', '19', '20'];
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const search = getSearchWith(searchParams, {
+      query: event.target.value || null,
+    });
+
+    setSearchParams(search);
+  };
```

Comment:
It seems like you are updating the searchParams state directly inside the event handler. This could potentially lead to some unexpected behavior because state updates in React are asynchronous. It might be better to create a new object with the updated state, and then use that to update your state.